source/base/Shmoose.cpp

```diff
@@ -123,7 +123,30 @@ void Shmoose::mainConnect(const QString &jid, const QString &pass)
 {
     persistence_->openDatabaseForJid(jid);
 
-    QString completeJid = jid + "/shmoose";
+    QString resourceName;
+    QString resourceId;
+
+    //generate unique resource name
+    resourceId = settings_->getResourceId();
+    if(resourceId.isEmpty())
+    {
+       const QString possibleCharacters("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz");
+       const int randomStringLength = 4; 
+
+       QString randomString;
+       for(int i=0; i<randomStringLength; ++i)
+       {
+           int index = qrand() % possibleCharacters.length();
+           QChar nextChar = possibleCharacters.at(index);
+           randomString.append(nextChar);
+       }
+
+       resourceId = randomString;
+       settings_->setResourceId(resourceId);
+    }
+
+    resourceName = QString("shmoose.") + resourceId;
+    QString completeJid = jid + "/" + resourceName;
```

Comment:
This is old code Most of it is now in System.cpp.

Remove this part and make sure that you don't mix up things in PR's.
You should create a new, clean branch from an up-to-date master for each PR.